scripts/analyze_deps.py

```diff
@@ -257,6 +256,8 @@ def render_report(output_path, report_context):
                     frozen[req_name] = [spec]
     except:
         print('Unable to open shared_requirements.txt, shared requirements have not been validated')
+        if inconsistent:
```

Comment:
This isn't quite right, because if the frozen file exists but is empty, we won't hit this except block, but neither will we set the exit code properly down below.

My suggestion would be to move this down to somewhere around L305, so the logic becomes:

```python
if frozen:
    ...
elif inconsistent:
    # When there are frozen reqs, we set the exitcode only if there
    # are inconsistencies that aren't covered by an override. If there
    # aren't any frozen reqs, any inconsistency should set the exitcode.
    exitcode = 1
```

The code also makes a little more sense that way, but I'd still add a comment as to why we're setting the exit code here.

Reply:
Got it. This makes sense.

Reply:
Pretty certain I got this right.